### PR TITLE
pub/sub rocketmq:upgrade with v2

### PR DESCRIPTION
**Pull request by zach030**
Signed-off-by: zach <zachchou016@gmail.com>

# Description
create a new PR for #1169 
upgrade rocketmq client with apache/rocketmq-client-go/v2.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by zach030** [comment]
@beiwei30 please review

**Comment by sduo** [comment]
@yaron2 Please, help

**Comment by yaron2** [comment]
@zach030 did you run this pub/sub locally and confirm that the following is true:

1. You can successfully publish messages to multiple topics
2. Multiple consumers can subscribe to multiple topics
3. Consumer groups work as intended (with multiple instances in a consumer group, only one instance gets a copy of a given message)

**Comment by zach030** [comment]
> @zach030 did you run this pub/sub locally and confirm that the following is true:
> 
> 1. You can successfully publish messages to multiple topics
> 2. Multiple consumers can subscribe to multiple topics
> 3. Consumer groups work as intended (with multiple instances in a consumer group, only one instance gets a copy of a given message)
> 4. At-least-once delivery works (messages get delivered in case a consumer crashes before ACK)

I wil run locally and post the result later~

File: pubsub/rocketmq/metadata.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
+// ------------------------------------------------------------
```
Comment (daixiang0) [inline pubsub/rocketmq/metadata.go:1]: Now we use apache license.

Comment (zach030) [reply pubsub/rocketmq/metadata.go:1]: change to: Licensed under the Apache License ?

**Comment by yaron2** [comment]
There's a minor linter failure, please take a look @zach030.

**Comment by zach030** [comment]
> There's a minor linter failure, please take a look @zach030.

fixed it! please run the workflow

File: pubsub/rocketmq/metadata.go (hunk 1 above)
Comment (daixiang0) [reply pubsub/rocketmq/metadata.go:1]: yes.

Comment (zach030) [reply pubsub/rocketmq/metadata.go:1]: > yes.

already changed

**Review by yaron2 — APPROVED** [review]
lgtm

**Comment by yaron2** [comment]
Merged! thanks @zach030. Can you please open a PR on `dapr/dapr` to put this inside `main.go` and open a docs issue for this component?

Thanks!

**Comment by zach030** [comment]
@yaron2 Thanks! https://github.com/dapr/dapr/pull/4016 is already opened for `rocketmq`, but it failed DCO test. Should I open a new PR for it?

**Comment by x-shadow-man** [comment]
> 
I'll try later. Thank for your mention! 👍

**Comment by x-shadow-man** [comment]
> @yaron2 Thanks! [dapr/dapr#4016](https://github.com/dapr/dapr/pull/4016) is already opened for `rocketmq`, but it failed DCO test. Should I open a new PR for it?

I'll try later. Thank for your mention!

**Pull request merged**